src/core/login-core.js

```diff
@@ -0,0 +1,219 @@
+import _ from "lodash";
+import { throwStatus } from "../util/express";
+import crypto from "crypto";
+
+import jwt from "jwt-simple";
+const { knex } = require("../util/database").connect();
+require("../init-env-variables");
+
+function tokenForUser(id) {
+  const timestamp = new Date().getTime();
+  return jwt.encode({ sub: id, iat: timestamp }, process.env.JWT_SECRET);
+}
+
+function sendTokenWithEmail(email, token) {
+  var sendgrid = require("sendgrid")(process.env.SENDGRID_API_KEY);
+  const request = sendgrid.emptyRequest({
+    method: "POST",
+    path: "/v3/mail/send",
+    body: {
+      personalizations: [
+        {
+          to: [
+            {
+              email: email
+            }
+          ],
+          subject: "Activation mail for your WhappuApp adminpanel account"
+        }
+      ],
+      from: {
+        email: process.env.EMAIL_ADDRESS
+      },
+      content: [
+        {
+          type: "text/plain",
+          value: process.env.FRONTEND_URI + "/activateaccount?token=" + token
+        }
+      ]
+    }
+  });
+  sendgrid
```

Comment:
This promise should probably be returned by the function, so that callers can handle it